archive/2025-10-30-claude-pirate-abusing-anthropic-s-file-api-for-data-exfiltration-embrace-the-red.md

```diff
@@ -0,0 +1,237 @@
+
+CancelConfirm
+
+Share
+
+Include playlist
+
+An error occurred while retrieving sharing information. Please try again later.
+
+[Why am I seeing this?](https://support.google.com/youtube/answer/9004474?hl=en)
+
+[Watch on](https://www.youtube.com/watch?v=2IoqhwqVI0s&embeds_referring_euri=https%3A%2F%2Fembracethered.com%2F)
+
+0:00
+
+0:00 / 11:17
+•Live
+
+•
+
+Hope it helps to highlight the scenario.
+
+### Demo Screenshots
+
+1. This is the attacker’s Anthropic Console before the attack.
+
+[![Claude Pirate - Attacker Console shows no files](https://embracethered.com/blog/images/2025/claude-pirate-step-1.png)](https://embracethered.com/blog/images/2025/claude-pirate-step-1.png)
+
+2. Now, we switch to the target user, and observe the last conversation from the chat history (this is what the demo will steal)
+
+[![Claude Pirate - Demo Data](https://embracethered.com/blog/images/2025/claude-pirate-step-2-zoom.png)](https://embracethered.com/blog/images/2025/claude-pirate-step-2.png)
+
+3. Now the user analyzes a malicious document from the attacker (indirect prompt injection, this could also come via an MCP server,…)
+
+[![Summarize this document](https://embracethered.com/blog/images/2025/claude-pirate-step-3-zoom.png)](https://embracethered.com/blog/images/2025/claude-pirate-step-3.png)
+
+4. AI Kill Chain at a glance: The exploit hijacks Claude and follows the adversaries instructions to grab private data, write it to the sandbox, and then calls the Anthropic File API to upload the file **to the attacker’s account using the attacker’s API key**
+
+[![AI Kill Chain Steps](https://embracethered.com/blog/images/2025/claude-pirate-step-4.png)](https://embracethered.com/blog/images/2025/claude-pirate-step-4.png)
+
+5. Attacker refreshes the Files view in their Console and the target’s uploaded file appears
+   [![Console upload of the file](https://embracethered.com/blog/images/2025/claude-pirate-step-5.png)](https://embracethered.com/blog/images/2025/claude-pirate-step-5.png)
+
+6. The attacker can now access the file, e.g. by using it in a chat
+
+
+[![Attacker views contents](https://embracethered.com/blog/images/2025/claude-pirate-step-6.png)](https://embracethered.com/blog/images/2025/claude-pirate-step-6.png)
+
+That’s it.
+
+## Responsible Disclosure
+
+I disclosed this to Anthropic via HackerOne on 10/25/2025 and the ticket was closed 1 hour later with the following statement:
+
+> “Thank you for your submission! Unfortunately, this particular issue you reported is explicitly out of scope as outlined in the [Policy Page](https://hackerone.com/anthropic).
+
+With further explanation that it is considered a model safety issue.
+
+However, I do not believe this is just a safety issue, but a security vulnerability with the default network egress configuration that can lead to exfiltration of your private information.
```

Comment:
100%. Anthropic did implement a hard boundary blocking most network calls so their intentions are clear. It's a classic bug.